compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1482,6 +1482,11 @@ private boolean isFunctionTypeAssignable(BInvokableType source, BInvokableType t
         return checkFunctionTypeEquality(source, target, unresolvedTypes, (s, t, ut) -> isAssignable(t, s, ut));
     }
 
+    private boolean isTypeParamAssignable(BType sourceParam, BType targetParam) {
+        return isAssignable(sourceParam, targetParam) ||
+                (isAssignable(sourceParam, symTable.xmlType) && isAssignable(targetParam, sourceParam));
```

Comment:
What is the reason to do this check? IMO this need to be done within the `isAssignable` function.

Reply:
This check is done for TypeParams. When checking type parameter assignability, following is used. `isAssignable(sourceParam, targetParam)`

In the case of xml we cannot use this check.
Consider the following code
xml x = xml `<a>a</a>`;
x.forEach(function (xml entry) {
});
Here the targetType is `xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text` and the source type is `xml`. Since an xml value cannot be assigned to ``xml:Element|xml:Comment|xml:ProcessingInstruction|xml:Text` the check for xml is made special case for type params. This cannot be done within the isAssignable function because it doesnt know that we are doing assignability check for TypeParamas.